jscomp/syntax/src/res_printer.ml

```diff
@@ -433,8 +432,14 @@ let printPolyVarIdent txt =
   (* numeric poly-vars don't need quotes: #644 *)
   if isValidNumericPolyvarNumber txt then Doc.text txt
   else
-    let txt = Ext_ident.unwrap_uppercase_exotic txt in
     match classifyIdentContent ~allowUident:true txt with
+    | UppercaseExoticIdent ->
```

Comment:
polyvar has a slightly different style.

`#Uppercase` is completely allowed, so just deleting `\` is not enough here

Reply:
Yes it does.

Maybe we can replace it to
```ml
      let len = String.length txt in
      Doc.text ((String.sub [@doesNotRaise]) txt 1 (len - 1))
```

Reply:
Either that, to a comment that that's the effect, is fine.
Just, the current code can be a bit confusing without additional explanation.

Reply:
Btw, I don't know if it is possible, but one could try to classify the ident content already at scanning time, instead of print time.
And any id that is classified to be exotic could be represented internally as `"id"`.
So the printer would need to do no special work.
And uppercase exotic ids would just be exotic ids that happen to start with capital letter -- nothing special.

Reply:
I had tried it within a previous PR to get as much work done from the scanner as possible, but it ultimately failed. I have no other ideas on how to do it while still using the existing parsetree as is. What do you mean `"id"`? Is there another place to mark it?

Reply:
Actually never mind, the scanner does not have the context that the printer has, such as when an uppercase identified is allowed.
Good to go for me.